### PR TITLE
feat: add metadata into oci manifest when 'kpm push'

### LINKED ISSUE #201: Enhancement : `kpm push` should add the metadata of the kcl package to the OCI manifest file.

### PR DESCRIPTION
**Pull request by zong-zhe**
#### 1. Does this PR affect any open issues?(Y/N) and add issue references (e.g. "fix #123", "re #123".):

- [ ] N
- [x] Y 

Fixed #201 

#### 2. What is the scope of this PR (e.g. component or file name):

pkg/oci/oci.go

#### 3. Provide a description of the PR(e.g. more details, effects, motivations or doc link):

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [x] Other

#### 4. Are there any breaking changes?(Y/N) and describe the breaking changes(e.g. more details, motivations or doc link):

- [x] N
- [ ] Y 

When the `kpm push` is executed, some of the data required by AH is filled into the annotation field of the OCI manifest conf.

#### 5. Are there test cases for these changes?(Y/N) select and add more details, references or doc links:

- [ ] Unit test
- [x] Integration test
- [ ] Benchmark (add benchmark stats below)
- [ ] Manual test (add detailed scripts or steps below)
- [ ] Other

### CODE REVIEW

File: pkg/oci/oci.go [hunk 1]
```diff
@@ -198,7 +200,7 @@ func (ociClient *OciClient) ContainsTag(tag string) (bool, *reporter.KpmEvent) {
 }
 
 // Push will push the oci artifacts to oci registry from local path
-func (ociClient *OciClient) Push(localPath, tag string) *reporter.KpmEvent {
+func (ociClient *OciClient) Push(localPath, tag string, annotations map[string]string) *reporter.KpmEvent {
```
Comment (Peefy) [inline pkg/oci/oci.go:203]: Put this annotation in the `PushWithAnnotations` function or using some options to do this feature.

File: pkg/oci/oci.go [hunk 2]
```diff
@@ -200,7 +201,12 @@ func (ociClient *OciClient) ContainsTag(tag string) (bool, *reporter.KpmEvent) {
 }
 
 // Push will push the oci artifacts to oci registry from local path
-func (ociClient *OciClient) Push(localPath, tag string, annotations map[string]string) *reporter.KpmEvent {
+func (ociClient *OciClient) Push(localPath, tag string) *reporter.KpmEvent {
+	return ociClient.PushWithOciManifest(localPath, tag, &opt.OciManifestOptions{})
+}
+
+// PushWithManifest will push the oci artifacts to oci registry from local path
+func (ociClient *OciClient) PushWithOciManifest(localPath, tag string, manifest_opts *opt.OciManifestOptions) *reporter.KpmEvent {
```
Comment (Peefy) [inline pkg/oci/oci.go:209]: s/manifest_opts/opts

File: pkg/oci/oci.go (hunk 1 above)
Comment (zong-zhe) [reply pkg/oci/oci.go]: fixed.

File: pkg/oci/oci.go (hunk 2 above)
Comment (zong-zhe) [reply pkg/oci/oci.go]: fixed.

**Review by Peefy — APPROVED** [review]
LGTM

**Pull request merged**